cody.el

```diff
@@ -640,24 +668,83 @@ The return value is appropiate for sending directly to the rpc layer."
              (end-pos (pos-parameters end)))
         (list :start beg-pos :end end-pos)))))
 
+(defun cody--handle-typing-while-suggesting (beg end len)
+  "Either recompute completion or dispel it, depending on the change.
+This can be called for either insertions or deletions, and we allow
+the completion suggestion overlay to remain active for certain instances
+of both. If it is a deletion, we recorded the deleted text in our
+before-change function.
+BEG and END are the region that changed, and LEN is its length."
+  (condition-case err
+      (if-let*
+          ;; TODO: We ought to get typing the completion prefix working,
```

Comment:
Can you just cancel the completion and request a new one with the typing?

IIRC VSCode does this typeover at the completion provider level, which recognizes a prefix of a cached completion matches the request; then it returns the suffix of the cached completion.

Or at least... that's how it worked at one point. There's been a lot of innovation in how completions are handled.